include/aws/common/array_list.inl

```diff
@@ -75,6 +75,14 @@ void aws_array_list_init_static(
     list->data = raw_array;
 }
 
+AWS_STATIC_IMPL
+bool aws_array_list_is_valid(const struct aws_array_list *AWS_RESTRICT list) {
+    size_t required_size;
+    return aws_mul_size_checked(list->length, list->item_size, &required_size) == AWS_OP_SUCCESS &&
```

Comment:
Thought: I found it easier to debug when I had the form
```
bool current_size_is_valid = list->current_size >= required_size; 
... 
return current_size_is_valid && ...
```
Not necessary to do this, but might make debugging easier.